src/tigerbeetle/repl.zig

```diff
@@ -0,0 +1,813 @@
+                    // Release allocation after every execution.
+                    var execution_arena = std.heap.ArenaAllocator.init(allocator);
+                    defer execution_arena.deinit();
+                    var statement = Parse.parse_statement(
+                        &execution_arena,
+                        statement_string,
+                    ) catch |err| {
+                        switch (err) {
+                            // These are parsing errors and since this
+                            // is not an interactive command, we should
+                            // exit immediately. Parsing error info
+                            // has already been emitted to stderr.
+                            error.BadIdentifier,
+                            error.BadOperation,
+                            error.BadValue,
+                            error.BadKeyValuePair,
+                            error.MissingEqualBetweenKeyValuePair,
+                            => std.os.exit(1),
+
+                            // An unexpected error for which we do
+                            // want the stacktrace.
+                            else => return err,
+                        }
+                    };
+                    try do_statement(repl, &execution_arena, statement);
+                }
+            } else {
+                try display_help();
+                std.os.exit(0);
+            }
+
+            while (!repl.event_loop_done) {
+                if (repl.request_done and repl.interactive) {
+                    // Release allocation after every execution.
+                    var execution_arena = std.heap.ArenaAllocator.init(allocator);
+                    defer execution_arena.deinit();
+                    try repl.do_repl(&execution_arena);
+                }
+                repl.client.tick();
+                try io.run_for_ns(constants.tick_ms * std.time.ns_per_ms);
+            }
+        }
+
+        fn create(
+            repl: *Repl,
+            arena: *std.heap.ArenaAllocator,
+            objects: []*Parse.ObjectSyntaxTree,
+            comptime T: type,
+            comptime object_type: Object,
+        ) !void {
+            if (objects.len == 0) {
+                try repl.fail("No {s}s to create.\n", .{@tagName(object_type)});
+                return;
+            }
+
+            var allocator = arena.allocator();
+            var batch = try std.ArrayList(T).initCapacity(allocator, objects.len);
+
+            for (objects) |object| {
+                batch.appendAssumeCapacity(@field(object, @tagName(object_type)));
+            }
+
+            assert(batch.items.len == objects.len);
+
+            // Submit batch.
+            try repl.send(
+                switch (object_type) {
+                    .account => .create_accounts,
+                    .transfer => .create_transfers,
+                },
+                std.mem.sliceAsBytes(batch.items),
+            );
+        }
+
```

Comment:
Hmm, I think this naming is confusing. "object" can refer an `ObjectSyntaxTree`, an account/transfer, or an enum of `{account,transfer}`.

How about any time we have a `ObjectSyntaxTree`, we call it an `object_tree`,
we use `Object` to refer to `tb.Account` or `tb.Transfer`, and remove the `object_type`/`Object` enum entirely. 

Maybe create a utility function that takes a `[]u8`, a `[]*Parse.ObjectSyntaxTree`, and `comptime Object: type` (`Account` or `Transfer`), and it writes the objecttrees to the buffer and returns the number of bytes written. (Pass it the `message` body).

Also I think `create`+`lookup` could be combined.

Reply:
Ok, I changed things around so that while the parser parses into ObjectSyntaxTree, it only stores the objects as bytes directly. This way there's only one copy from the arguments into the message.

Maybe that's more confusing, let me know!